cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -363,53 +363,33 @@ folly::Future<DescriptorItem> LocalVersionedEngine::get_descriptor(
     const auto key = std::move(k);
     return store()->read(key)
     .thenValue([](auto&& key_seg_pair) -> DescriptorItem {
-        auto key_seg = std::move(std::get<0>(key_seg_pair));
-        auto seg = std::move(std::get<1>(key_seg_pair));
-        auto timeseries_descriptor = seg.has_metadata() ? std::make_optional<google::protobuf::Any>(*seg.metadata()) : std::nullopt;
-        auto start_index = seg.column(position_t(index::Fields::start_index)).type().visit_tag([&](auto column_desc_tag) -> std::optional<NumericIndex> {
-            using ColumnDescriptorType = std::decay_t<decltype(column_desc_tag)>;
-            using ColumnTagType =  typename ColumnDescriptorType::DataTypeTag;
-            if (seg.row_count() == 0) {
-                return std::nullopt;
-            } else if constexpr (is_numeric_type(ColumnTagType::data_type)) {
-                std::optional<NumericIndex> start_index;
-                auto column_data = seg.column(position_t(index::Fields::start_index)).data();
-                while (auto block = column_data.template next<ColumnDescriptorType>()) {
-                    auto ptr = reinterpret_cast<const NumericIndex *>(block.value().data());
-                    const auto row_count = block.value().row_count();
-                    for (auto i = 0u; i < row_count; ++i) {
-                        auto value = *ptr++;
-                        start_index = start_index.has_value() ? std::min(*start_index, value) : value;
-                    }
+        auto key = to_atom(std::move(key_seg_pair.first));
+        auto seg = std::move(key_seg_pair.second);
+        auto tsd = std::make_optional<arcticdb::proto::descriptors::TimeSeriesDescriptor>();
+        if (seg.has_metadata()) {
+            seg.metadata()->UnpackTo(&(*tsd));
+        }
+        std::optional<timestamp> start_index;
+        std::optional<timestamp> end_index;
+        if (seg.row_count() > 0) {
+            const auto& start_index_column = seg.column(position_t(index::Fields::start_index));
+            details::visit_type(start_index_column.type().data_type(), [&start_index_column, &start_index](auto column_desc_tag) {
+                using type_info = ScalarTypeInfo<decltype(column_desc_tag)>;
+                if constexpr (is_time_type(type_info::data_type)) {
+                    start_index = start_index_column.template scalar_at<timestamp>(0);
```

Comment:
Was there a reason not to use start_value_for_segment and end_value_for_segment from index.hpp?

Reply:
This is an index key, not a data key